test/whitespace/functions.unit

```diff
@@ -165,4 +165,12 @@ main() {
 >>> generic function typed parameter
 function(int   foo  <  T  ,S >(T t, S s)) {}
 <<<
-function(int foo<T, S>(T t, S s)) {}
\ No newline at end of file
+function(int foo<T, S>(T t, S s)) {}
+>>> decorator-style function
+function() => decorator(() {
+  body;
+});
+<<<
+function() => decorator(() {
+  body;
+});
```

Comment:
Don't forget the first bullet point: the function declaration itself wraps, not the decorated one:

```dart
function(long, stuff, here) => ...
```